torch/lib/c10d/TcpStore.cpp

```diff
@@ -0,0 +1,326 @@
+#include "TcpStore.hpp"
+
+#include <poll.h>
+#include <unistd.h>
+#include <system_error>
+
+namespace c10d {
+
+namespace {
+
+enum class QueryType : std::uint8_t {
+  SET,
+  GET,
+  ADD,
+  CHECK
+};
+
+enum class CheckResponseType : std::uint8_t {
+  READY,
+  NOT_READY
+};
+
+} // anonymous namespace
+
+// TCPStoreDaemon class methods
+// Simply start the daemon thread
+TCPStoreDaemon::TCPStoreDaemon(int storeListenSocket) :
+  storeListenSocket_(storeListenSocket)
+{
+  daemonThread_ = std::thread(&TCPStoreDaemon::run, this);
+}
+
+TCPStoreDaemon::~TCPStoreDaemon() {
+
+  // Stop the run
+  stop();
+
+  // Join the thread
+  join();
+
+  // Close unclosed sockets
+  for (auto socket : sockets_) {
+    if (socket != -1) {
+      ::close(socket);
+    }
+  }
+  // Now close the rest control pipe
+  for (auto fd : controlPipeFd_) {
+    if (fd != -1) {
+      ::close(fd);
+    }
+  }
+}
+
+void TCPStoreDaemon::join() {
+  daemonThread_.join();
+}
+
+void TCPStoreDaemon::run() {
+
+  // Create the control pipe
+  controlPipeFd_ = std::vector<int>{-1, -1};
```

Comment:
You know it will be of size 2, so you can use a std::array<int, 2> instead, and initialize in the constructor

Reply:
done